services/QuillLMS/app/services/quill_big_query/transformer.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module QuillBigQuery
+  class Transformer
+    attr_reader :field_to_lambda, :array_of_hashes
+
+    class UnsupportedBigQueryType < StandardError; end
+
+    def initialize(fields, array_of_hashes)
+      @field_to_lambda = self.class.build_field_to_lambda_lookup(fields)
+      @array_of_hashes = array_of_hashes
+    end
+
+    LAMBDA_IDENTITY     = ->(val) { val }
+    LAMBDA_TO_BOOLEAN   = ->(val) { val == 'true' }
+    LAMBDA_TO_DATETIME  = ->(val) { DateTime.parse(val) }
+    LAMBDA_TO_F         = ->(val) { val.to_f }
+
+    TYPE_TO_LAMBDA_LOOKUP = {
+      'ARRAY'       => LAMBDA_IDENTITY,
+      'BIGNUMERIC'  => LAMBDA_IDENTITY,     # '5.2e+37'
+      'BOOLEAN'     => LAMBDA_TO_BOOLEAN,   # 'true'
+      'BOOL'        => LAMBDA_TO_BOOLEAN,
+      'BYTES'       => LAMBDA_IDENTITY,
+      'DATE'        => LAMBDA_TO_DATETIME,  # '2022-01-01'
+      'DATETIME'    => LAMBDA_TO_DATETIME,  # '2022-05-06T17:08:44.915791'
+      'FLOAT'       => LAMBDA_TO_F,
+      'FLOAT64'     => LAMBDA_TO_F,
+      'GEOGRAPHY'   => LAMBDA_IDENTITY,
+      'INTEGER'     => LAMBDA_TO_F,
+      'INT64'       => LAMBDA_TO_F,
+      'NUMERIC'     => LAMBDA_TO_F,
+      'STRING'      => LAMBDA_IDENTITY,
+      'STRUCT'      => LAMBDA_IDENTITY,
+      'TIME'        => LAMBDA_TO_DATETIME,
+      'TIMESTAMP'   => LAMBDA_TO_DATETIME   # '2021-05-01 21:32:45'
+    }
+
+    def self.build_field_to_lambda_lookup(fields)
```

Comment:
Is this method meant to be used elsewhere?

Otherwise, why not make a private instance method and remove the parameter passing.

Reply:
I consider the complexity of the two options equal, and thus its a stylistic choice:

| current | proposed |
|--------|--------|
| pure function  | impure |
| public | private |
| needs param | does not|
| | needs initialization, for `fields` |

Happy to discuss further if you feel strongly about it, however.